feature(simport) TypeScript and utilities

jlarmstrongiv opened the pull request:
<!--
Thank you for making pull request. Please fill in the template below. If unsure
about something, just do as best as you're able.
-->

- [x] commit message named according to [Contributing Guide](https://github.com/coderaiser/cloudcmd/blob/master/CONTRIBUTING.md "Contributting Guide")
- [ ] `npm run codestyle` is OK (unable to run codestyle on my machine)
- [x] `npm test` is OK

Added TypeScript typings and utility functions.  

Since there is only a single typings file for both ESM and CommonJS, I ensured both had corresponding functions.  I also added standalone utilities for `__filename` and `__dirname`.  

One nice benefit of using `createCommons` in CommonJS is that it makes refactoring to ESM in the future substantially easier.

jlarmstrongiv committed:
feature(simport) TypeScript and utilities

coderaiser commented:
This is very good!

coderaiser commented on index.d.ts:
````diff
@@ -0,0 +1,93 @@
+import {createDirname} from 'simport';
+const __dirname = createDirname(import.meta.url);
+```
+*/
````
We can put this to `README.md` instead of comments in types.

jlarmstrongiv replied:
The comments in types show up in places like VSCode autocomplete/info boxes.  I find them to be very helpful 😄 

It should probably go in the `README.md` too

jlarmstrongiv replied:
I updated the README.md with the individual utility function examples 👍

jlarmstrongiv committed:
docs(readme) add utility examples

jlarmstrongiv committed:
chore(package) remove package-lock.json

jlarmstrongiv commented:
Hi @coderaiser 👋 

I added examples to the documentation and removed the `package-lock.json`.

The tests and check-dts would likely need to be written for both cjs and mjs files.  I think https://www.npmjs.com/package/tsd may be a better fit for testing the types.

Unfortunately, I am closing this PR since I am unable to make those additions in the near future.  Thanks for reviewing it though!
